src/bench/bench.cpp

```diff
@@ -8,30 +8,35 @@
 #include <assert.h>
 #include <iostream>
 #include <iomanip>
-#include <sys/time.h>
+
+template <typename Clock>
+static constexpr const char* get_clock_precision()
+{
+    return std::ratio_less<typename Clock::period, std::pico>::value ? "< picosecond" :
+            std::ratio_equal<typename Clock::period, std::pico>::value ? "picosecond" :
+            std::ratio_equal<typename Clock::period, std::nano>::value ? "nanosecond" :
+            std::ratio_equal<typename Clock::period, std::micro>::value ? "microsecond" :
+            std::ratio_equal<typename Clock::period, std::milli>::value ? "millisecond" :
+            "> millisecond";
+}
 
 benchmark::BenchRunner::BenchmarkMap &benchmark::BenchRunner::benchmarks() {
     static std::map<std::string, benchmark::BenchFunction> benchmarks_map;
     return benchmarks_map;
 }
 
-static double gettimedouble(void) {
-    struct timeval tv;
-    gettimeofday(&tv, nullptr);
-    return tv.tv_usec * 0.000001 + tv.tv_sec;
-}
-
 benchmark::BenchRunner::BenchRunner(std::string name, benchmark::BenchFunction func)
 {
     benchmarks().insert(std::make_pair(name, func));
 }
 
 void
-benchmark::BenchRunner::RunAll(double elapsedTimeForOne)
+benchmark::BenchRunner::RunAll(benchmark::duration elapsedTimeForOne)
 {
     perf_init();
     std::cout << "#Benchmark" << "," << "count" << "," << "min" << "," << "max" << "," << "average" << ","
-              << "min_cycles" << "," << "max_cycles" << "," << "average_cycles" << "\n";
+              << "min_cycles" << "," << "max_cycles" << "," << "average_cycles. " << "Clock precision: "
```

Comment:
Please use a `,`, to separate here, instead of a dot, otherwise programs will get confused re: CSV parsing.